analyzers/tests/SonarAnalyzer.UnitTest/Helpers/CSharpSymbolUsageCollectorTest.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.UnitTest.TestFramework;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class CSharpSymbolUsageCollectorTest
+    {
+        [TestMethod]
+        public void VerifyUsagesBeingCollectedOnMatchingSyntaxNodes()
+        {
+            const string code1 = @"
+public partial class Sample
+{
+    private int Field = 42;
+
+    public int Foo(int arg)
+    {
+        Field += arg;
+        return Field;
+    }
+}";
+            const string code2 = @"
+public partial class Sample
+{
+    public int Bar()
+    {
+        return Field;
+    }
+}";
+            var compilation1 = SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp, createExtraEmptyFile: false)
```

Comment:
same here: I suggest to use "firstCompilation" and "secondCompilation" (and "firstTree" and "secondTree") because it's easier do differentiate the names